server.js

```diff
@@ -1,38 +1,117 @@
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import mongoose from 'mongoose';
+
+import booksData from './data/books.json';
+
+// Code for setting up our Mongo database
+const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
+
+// Create Book model to start populating our database
+const Book = mongoose.model('Book', {
+  bookID: Number,
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: String,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+});
+
+// Function to start seeding our Database, will only run if
+// RESET_DB environment variable is present and is true
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    // Starts by deleting any pre-existing Book objects to prevent duplicates
+    await Book.deleteMany({});
+    
+    // Creates a new Book instance for each book in the booksData
+    booksData.forEach((bookData) => {
+      new Book(bookData).save();
+    })
+  };
+  seedDatabase();
+};
+
+// Defines the port the app will run on
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
 
-// Start defining your routes here
+// ROUTES
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Hello world, welcome to Vane Bookish API, now powered by MongoDB!');
+});
+
+// Route to get all the books in the database
+app.get('/books', async (req, res) => {
+  const allBooks = await Book.find();
+  res.json(allBooks);
+});
+
+// Route to get a single book based on its ID
+app.get('/books/book/:bookID', async (req, res) => {
+  try {
+    const singleBook = await Book.findOne({ bookID: req.params.bookID });
+
+    if (singleBook) {
+      res.json(singleBook);
+    } else {
+      // Error when the book id format is valid, but still no book is found with that id
+      res.status(404).json("Sorry, no books found with that ID :(");
+    }
+  } catch (err) {
+    // error when the book id format is wrong, an invalid book id is entered
+    res.status(400).json({ error: "Invalid Book ID, double check book id value" });
+  }
```

Comment:
Impressive that you have two error messages depending upon what the user has entered. Smart to think this way to cover all the bases 👌